nixos/modules/services/audio/spotifyd.nix

```diff
@@ -23,17 +23,14 @@ in
   };
 
   config = mkIf cfg.enable {
-    systemd.services.spotifyd = {
+    systemd.user.services.spotifyd = {
```

Comment:
I still think using the service file from the package would be better:

```patch
diff --git a/nixos/modules/services/audio/spotifyd.nix b/nixos/modules/services/audio/spotifyd.nix
index 61058a2f419..102259f8b0d 100644
--- a/nixos/modules/services/audio/spotifyd.nix
+++ b/nixos/modules/services/audio/spotifyd.nix
@@ -24,15 +24,14 @@ in
 
   config = mkIf cfg.enable {
     systemd.user.services.spotifyd = {
-      wantedBy = [ "multi-user.target" ];
-      after = [ "network-online.target" "sound.target" ];
-      description = "spotifyd, a Spotify playing daemon";
       serviceConfig = {
+        # Overriding config file location
+        # Keep in sync with ${spotifyd}/lib/systemd/user/spotifyd.service
         ExecStart = "${pkgs.spotifyd}/bin/spotifyd --no-daemon --config-path ${spotifydConf}";
-        Restart = "always";
-        RestartSec = 12;
       };
     };
+
+    systemd.packages = [ pkgs.spotifyd ];
   };
 
   meta.maintainers = [ maintainers.anderslundstedt ];
diff --git a/pkgs/applications/audio/spotifyd/default.nix b/pkgs/applications/audio/spotifyd/default.nix
index ad84d5298d4..266a27ccf13 100644
--- a/pkgs/applications/audio/spotifyd/default.nix
+++ b/pkgs/applications/audio/spotifyd/default.nix
@@ -39,6 +39,14 @@ in rustPlatform.buildRustPackage rec {
 
   doCheck = false;
 
+  postPatch = ''
+    substituteInPlace contrib/spotifyd.service --replace "/usr/bin" "$out/bin"
+  '';
+
+  postInstall = ''
+    install -Dm644 contrib/spotifyd.service $out/lib/systemd/user/spotifyd.service
+  '';
+
   meta = with stdenv.lib; {
     description = "An open source Spotify client running as a UNIX daemon";
     homepage = "https://github.com/Spotifyd/spotifyd";
```

Reply:
don't we want `wantedBy`, `after` and `description` ? why do we care to keep in sync with the package's version?

Reply:
Well, if we use the file from the package, there already are those fields. The assumption is that the upstream file contains the correct options in `ExecStart` so we should sync that.

Reply:
oh I didn't know systemd somehow merges config options from multiple service definitions? maybe I'm confused on how that works.